markets/perps-market/contracts/storage/GlobalPerpsMarket.sol

```diff
@@ -73,13 +73,14 @@ library GlobalPerpsMarket {
         Data storage self
     ) internal view returns (uint128 rate, uint256 delegatedCollateralValue, uint256 lockedCredit) {
         uint256 withdrawableUsd = PerpsMarketFactory.totalWithdrawableUsd();
-        delegatedCollateralValue = withdrawableUsd - totalCollateralValue(self);
+        int256 delegatedCollateralValueAsInt = withdrawableUsd.toInt() -
+            totalCollateralValue(self).toInt();
         lockedCredit = minimumCredit(self);
-        if (delegatedCollateralValue == 0) {
+        if (delegatedCollateralValueAsInt <= 0) {
             return (0, 0, lockedCredit);
         }
 
-        rate = lockedCredit.divDecimal(delegatedCollateralValue).to128();
+        rate = lockedCredit.divDecimal(delegatedCollateralValueAsInt.toUint()).to128();
```

Comment:
I dont think we need to re-uInt it 😅  since we already caught he negative case?

Reply:
need to make it uint to divide by another uint